cmd/kubeturbo/app/kubeturbo_builder.go

```diff
@@ -1009,3 +1011,47 @@ func reviewSCCAccess(namespace string, kubeClient kubernetes.Interface) bool {
 
 	return true
 }
+
+func WatchConfigMap() {
+	//Check if the file /etc/kubeturbo/turbo-autoreload.config exists
+	autoReloadConfigFilePath := "/etc/kubeturbo"
+	autoReloadConfigFileName := "turbo-autoreload.config"
+
+	viper.AddConfigPath(autoReloadConfigFilePath)
+	viper.SetConfigType("json")
+	viper.SetConfigName(autoReloadConfigFileName)
+	for {
+		verr := viper.ReadInConfig()
+		if verr == nil {
+			break
+		} else {
+			glog.V(4).Infof("Can't read the autoreload config file %s/%s due to the error: %v, will retry in 3 seconds", autoReloadConfigFilePath, autoReloadConfigFileName, verr)
+			time.Sleep(3 * time.Second)
```

Comment:
I don't think this will be a common case. Suggest increasing the interval to 60 seconds.

Reply:
When the user adds this dynamic logging level the first time,  if we change this interval to 60s, the user may need to wait for around 2 minutes to see the logging level is finally changed, 1 minute on this waiting, 1 minute waiting for the file change to be reflected inside the container. Is it a bit longer?

Reply:
I am not sure how viper works, but with inotify we cannot install a watch on a non existent file. 
We first check if the actual file exists, if it does, we load it and install a watch on it. If not, we do install the fsnotify watch on the parent directory first, which in effect means when a new file is created we get a notification. Once we have a new file event for our file, we can load that and start watching for that file instead.


If viper is handling the above logic internally, for the case you seem to be handling above, if the file exists at start, but is malformed, we can simply post an error and move on. Whenever the file is corrected next, our notify watch should catch the update event and we then proceed with that. I don't see a point in retry.
What I mean is all this handling should ideally be in you callback `updateConfig()` below.

Reply:
@irfanurrehman  Thanks for reviewing my code.  I believe you have seen the loop I use to check if the config file is ready for watching or not.  Just as you expect, I keep checking the config file until it's pared well and then add a watch on it.  If the file doesn't exist or is not parsed well, the code will be waiting there.

Reply:
@ading1977  I change the check interval to `30s`, does that make sense to you?